pkg/proxy/healthcheck/proxy_health.go

```diff
@@ -70,34 +71,31 @@ type ProxyHealthServer struct {
 	httpFactory httpServerFactory
 	clock       clock.Clock
 
+	nodeManager *proxy.NodeManager
+
 	addr          string
 	healthTimeout time.Duration
 
 	lock                   sync.RWMutex
 	lastUpdatedMap         map[v1.IPFamily]time.Time
 	oldestPendingQueuedMap map[v1.IPFamily]time.Time
-	nodeEligible           bool
 }
 
 // NewProxyHealthServer returns a proxy health http server.
-func NewProxyHealthServer(addr string, healthTimeout time.Duration) *ProxyHealthServer {
-	return newProxyHealthServer(stdNetListener{}, stdHTTPServerFactory{}, clock.RealClock{}, addr, healthTimeout)
+func NewProxyHealthServer(addr string, healthTimeout time.Duration, nodeManager *proxy.NodeManager) *ProxyHealthServer {
+	return newProxyHealthServer(stdNetListener{}, stdHTTPServerFactory{}, clock.RealClock{}, addr, healthTimeout, nodeManager)
 }
 
-func newProxyHealthServer(listener listener, httpServerFactory httpServerFactory, c clock.Clock, addr string, healthTimeout time.Duration) *ProxyHealthServer {
+func newProxyHealthServer(listener listener, httpServerFactory httpServerFactory, c clock.Clock, addr string, healthTimeout time.Duration, nodeManager *proxy.NodeManager) *ProxyHealthServer {
 	return &ProxyHealthServer{
-		listener:      listener,
-		httpFactory:   httpServerFactory,
-		clock:         c,
-		addr:          addr,
-		healthTimeout: healthTimeout,
-
+		listener:               listener,
+		httpFactory:            httpServerFactory,
+		clock:                  c,
+		addr:                   addr,
+		healthTimeout:          healthTimeout,
+		nodeManager:            nodeManager,
```

Comment:
there was a blank line before `lastUpdatedMap` before and if you leave that there you won't have to reindent everything